### PR TITLE
fix: Remove previous RELATED_IMAGES before adding new ones

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
Remove previous RELATED_IMAGES before adding new ones to `config/manager/manager.yaml`

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20171

### How to test this PR?

Run twice, make sure that content of the `config/manager/manager.yaml` is not changes after the second try.
`./olm/addDigests.sh -w . -s bundle/stable/eclipse-che-preview-openshift/manifests/che-operator.clusterserviceversion.yaml -o config/manager/manager.yaml -t 7.32.0`

### PR Checklist
          
[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by flacatus — APPROVED** [review]

File: olm/addDigests.sh [hunk 1]
```diff
@@ -124,10 +124,22 @@ yq -riY "( .spec.relatedImages ) += [${RELATED_IMAGES}]" ${CSV_FILE}
...
 
 if [[ ${OPERATOR_FILE} ]]; then
+  # delete previous `RELATED_IMAGES`
+  envVarLength=$(cat "${OPERATOR_FILE}" | yq -r ".spec.template.spec.containers[0].env | length")
+  i=0
+  while [ "${i}" -lt "${envVarLength}" ]; do
+    envVarName=$(cat "${OPERATOR_FILE}" | yq -r '.spec.template.spec.containers[0].env['${i}'].name')
+    if [[ ${envVarName} =~ plugin_registry_image ]] || [[ ${envVarName} =~ devfile_registry_image ]]; then
+      yq -riY 'del(.spec.template.spec.containers[0].env['${i}'])' ${OPERATOR_FILE}
+      i=$((i-1))
```
Comment (mmorhun) [inline olm/addDigests.sh:137]: 1. I'd remove this decrement and put the following increment into else block.
2. Shouldn't you change the total number of env vars too?

File: olm/addDigests.sh [hunk 2]
```diff
@@ -124,10 +124,22 @@ yq -riY "( .spec.relatedImages ) += [${RELATED_IMAGES}]" ${CSV_FILE}
...
-
 echo "[INFO] CSV updated: ${CSV_FILE}"
 
 if [[ ${OPERATOR_FILE} ]]; then
+  # delete previous `RELATED_IMAGES`
+  envVarLength=$(cat "${OPERATOR_FILE}" | yq -r ".spec.template.spec.containers[0].env | length")
+  i=0
+  while [ "${i}" -lt "${envVarLength}" ]; do
+    envVarName=$(cat "${OPERATOR_FILE}" | yq -r '.spec.template.spec.containers[0].env['${i}'].name')
+    if [[ ${envVarName} =~ plugin_registry_image ]] || [[ ${envVarName} =~ devfile_registry_image ]]; then
```
Comment (mmorhun) [inline olm/addDigests.sh:135]: If you preserve only two env vars, does it make sense recreate whole list with these two?

Comment (tolusha) [reply olm/addDigests.sh:135]: That's a regexp

File: olm/addDigests.sh (hunk 1 above)
Comment (tolusha) [reply olm/addDigests.sh:137]: 1. I've thought about it but this approach sounds more explicit
2. It does not matter, there are no errors, `yq` returns `null` as a string

File: olm/addDigests.sh (hunk 2 above)
Comment (mmorhun) [reply olm/addDigests.sh:135]: Will it ever preserve more than 2 env vars?

File: olm/addDigests.sh (hunk 1 above)
Comment (mmorhun) [reply olm/addDigests.sh:137]: 1. I have opposite opinion, but ok.
2. That's logically wrong.

File: olm/addDigests.sh (hunk 2 above)
Comment (tolusha) [reply olm/addDigests.sh:135]: Sure

File: olm/addDigests.sh (hunk 1 above)
Comment (tolusha) [reply olm/addDigests.sh:137]: 2. It is not a c++ or java. We have to processed all records, other doesn't matter

File: olm/addDigests.sh (hunk 2 above)
Comment (mmorhun) [reply olm/addDigests.sh:135]: I see, due to some reason I missed the negations, sorry.

File: olm/addDigests.sh (hunk 1 above)
Comment (mmorhun) [reply olm/addDigests.sh:137]: That's a source of errors. If `yq` behavior changes or someone add another line that doesn't tolerate index error... but up to you...

**Review by mmorhun — APPROVED** [review]

**Pull request merged**